src/ex2/Uilogic.js

```diff
@@ -0,0 +1,444 @@
+    this.handleCountTask();
+    if (taskObject.isAnimated) {
+      this.AnimateListAdd(li);
+      this.updateTask(taskObject.id, taskObject.type, taskObject.name, taskObject.datetime, false)
+    }
+    this.DisplayInvisibleButtons(); // display sort and clear all buttons
+    li.addEventListener("click", ({ target }) => {
+      this.CheckedTask(target, false); // mark task for complete status
+    });
+    this.CreateRemoveItemFromList(li, taskList, taskObject.id);
+    this.addCalendarIcon(li, taskObject.datetime);
+    const pokemonObj = this.getPokemonObjectByName(taskList, taskObject.name);
+    if (pokemonObj !== null && index === 0) {
+      this.addPokemonImage(pokemonObj);
+    }
+    if (taskObject.type == TYPE_FINISHED) {
+      this.CheckedTask(li, true)
+    }
+    else if (taskObject.type == TYPE_DEADLINE_FINISHED) {
+      li.classList.add('markDeadline');
+    }
+
+  }
+
+  randomIntFromInterval(min, max) { // min and max included 
+    return Math.floor(Math.random() * (max - min + 1) + min)
+  }
+
+  CheckedTask(target, withSort) {
+    console.log(target.tagName);
+    if (target.tagName == "INPUT" || target.tagName == "BUTTON") {
+      return;
+    }
+    // Add a "checked" symbol when clicking on a list item
+    if (!target.classList.contains('checked')) {
+      const id = target.getAttribute('taskId')
+      let taskObject = this.getTask(id);
+      this.updateTask(taskObject.id, TYPE_FINISHED, taskObject.name, taskObject.datetime, false)
+      if (!withSort) {
+        this.rotateImage();
+      }
+
+    }
+    else {
+      let taskObject = this.getTask(target.getAttribute('taskId'));
+      this.updateTask(taskObject.id, TYPE_NOTCOMPLETE, taskObject.name, taskObject.datetime, false)
+    }
+    target.classList.toggle('checked');
+  }
+
+
+  sortList() {
+    this.bindTaskList(true);
+  }
+
+  clearList() {
+    this.taskList = [];
+    this.VanishvisibleButtons();
+    this.DisplayEmptyTaskFloatMassage();
+    document.getElementById("my-ul").innerHTML = "";
+    this.handleCountTask();
+  }
+
+  VanishvisibleButtons() {
+    const sortButton = document.getElementById("sort-btn");
+    const clearButton = document.getElementById("clear-btn");
+    sortButton.style.visibility = "hidden";
+    clearButton.style.visibility = "hidden";
+    this.VanishDropdown();
+  }
+  DisplayInvisibleButtons() {
+    const sortButton = document.getElementById("sort-btn");
+    const clearButton = document.getElementById("clear-btn");
+    sortButton.style.visibility = "visible";
+    clearButton.style.visibility = "visible";
+  }
+
+  AnimateListItemDelete(li) {
+    const spinning = [
+      { transform: 'rotate(0) scale(1)' },
+      { transform: 'rotate(0deg) scale(0)' }
+    ];
+    const timing = {
+      duration: 2000,
+      iterations: 1,
+    }
+    li.animate(spinning, timing);
+  }
+
+  AnimateListAdd(li) {
+    const spinning = [
+      { transform: 'rotate(0deg) scale(0)' },
```

Comment:
consider: prefer having the animations as a css class. having css as javascript is less efficient and less readable. I know it may be big effort to refactor it, so it's up to you.